cmd/discovery/client/config.go

```diff
@@ -0,0 +1,34 @@
+package client
+
+import (
+	"github.com/kairos-io/kairos/pkg/config"
+	kconfig "github.com/kairos-io/kcrypt/pkg/config"
+)
+
+type Client struct {
+	Config Config
+}
+
+type Config struct {
+	Kcrypt struct {
+		Server    string `yaml:"challenger_server,omitempty"`
+		NVIndex   string `yaml:"nv_index,omitempty"`
+		CIndex    string `yaml:"c_index,omitempty"`
+		TPMDevice string `yaml:"tpm_device,omitempty"`
+	}
+}
+
+func unmarshalConfig() (Config, error) {
+	var result Config
+
+	c, err := config.Scan(config.Directories(kconfig.ConfigScanDirs...), config.NoLogs)
```

Comment:
I was trying to re-use the `GetConfiguration` logic which is also tested on the `kcrypt` side. From the user's perspective this is all "kcrypt configuration", with "kcrypt" being just a feature. There are multiple binaries involved but they share the same configuration block and I think it should be parsed using the same code.